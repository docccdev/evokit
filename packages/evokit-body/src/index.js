import { createBlock } from 'evokit';

export const Body = createBlock('body', 'body', [
    'background',
    'color',
    'display',
    'indent',
    'size',
    ['overflow', [
        'overflow-x',
        'overflow-y',
    ]],
]);

export const BodySection = createBlock('div', 'body__section', ['display']);

Body.Section = BodySection;
