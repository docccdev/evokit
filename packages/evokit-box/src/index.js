// 3.2.0-beta.1
import { createBlock } from 'evokit';

export const Box = createBlock('div', 'box', [
    'align',
    'background',
    'display',
    'height',
    'opacity',
    'width',
    ['position', [
        'position',
        'place',
        'zindex',
    ]],
    ['border', [
        'border-color',
        'border-width',
        'border-style',
    ]],
    ['border-top', [
        'border-color-top',
        'border-width-top',
        'border-style-top',
    ]],
    ['border-right', [
        'border-color-right',
        'border-width-right',
        'border-style-right',
    ]],
    ['border-bottom', [
        'border-color-bottom',
        'border-width-bottom',
        'border-style-bottom',
    ]],
    ['border-left', [
        'border-color-left',
        'border-width-left',
        'border-style-left',
    ]],
    ['border-color', [
        ['border-color-top', 'border-color-bottom'],
        ['border-color-right', 'border-color-left'],
        'border-color-bottom',
        'border-color-left',
    ]],
    ['border-width', [
        ['border-width-top', 'border-width-bottom'],
        ['border-width-right', 'border-width-left'],
        'border-width-bottom',
        'border-width-left',
    ]],
    ['border-style', [
        ['border-style-top', 'border-style-bottom'],
        ['border-style-right', 'border-style-left'],
        'border-style-bottom',
        'border-style-left',
    ]],
    ['margin', [
        ['margin-top', 'margin-bottom'],
        ['margin-right', 'margin-left'],
        'margin-bottom',
        'margin-left',
    ]],
    ['overflow', [
        'overflow-x',
        'overflow-y',
    ]],
    ['padding', [
        ['padding-top', 'padding-bottom'],
        ['padding-right', 'padding-left'],
        'padding-bottom',
        'padding-left',
    ]],
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
    // DEPRECATED
    'margin-lr',
    'margin-tb',
    'padding-lr',
    'padding-tb',
    'round-bottom',
    'round-left',
    'round-right',
    'round-top',
    'border-lr',
    'border-tb',
]);
