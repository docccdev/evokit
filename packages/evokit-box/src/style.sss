@import './mixins.sss'

@custom-media --ek-box-media-small only screen and (min-width: 480px)
@custom-media --ek-box-media-medium only screen and (min-width: 768px)
@custom-media --ek-box-media-large only screen and (min-width: 960px)
@custom-media --ek-box-media-wide only screen and (min-width: 1200px)
@custom-media --ek-box-media-huge only screen and (min-width: 1400px)

:root
    // prop 'box-padding', 'box-margin'
    --ek-box-indent-xxs: 5px
    --ek-box-indent-xs: 10px
    --ek-box-indent-s: 15px
    --ek-box-indent-m: 20px
    --ek-box-indent-l: 25px
    --ek-box-indent-xl: 30px
    --ek-box-indent-xxl: 35px
    // prop 'box-round'
    --ek-box-round-xxs: 2px
    --ek-box-round-xs: 4px
    --ek-box-round-s: 6px
    --ek-box-round-m: 8px
    --ek-box-round-l: 10px
    --ek-box-round-xl: 12px
    --ek-box-round-xxl: 14px
    // prop 'box-border-width'
    --ek-box-border-width-xxs: 1px
    --ek-box-border-width-xs: 2px
    --ek-box-border-width-s: 3px
    --ek-box-border-width-m: 4px
    --ek-box-border-width-l: 5px
    --ek-box-border-width-xl: 6px
    --ek-box-border-width-xxl: 7px

.box
    display: block
    box-sizing: border-box
    border-width: 0
    border-style: solid

    &:after
        clear: both

    &:after, &:before
        content: ''
        display: table

/**
 * Align:
 *
 *  - box_align_none
 *  - box_align_left
 *  - box_align_center
 *  - box_align_right
 *
 **/

@each $value in (none, left, center, right)
    @mixin media .box_align_$(value)
        @if $value == center
            float: none
            display: inline-block
            position: relative
            left: 50%
            transform: translateX(-50%)
        @else
            float: $value
            display: block
            position: static
            left: 0
            transform: translateX(0)

/**
 * Display:
 *
 *  - box_display_*
 *
 **/

@each $value in (none, inline, inline-block, block)
    @mixin media .box_display_$(value)
        display: $value

        @if $value == inline
            &:after, &:before
                content: none

/**
 * Position:
 *
 *  - box_position_*
 *
 **/

@each $value in (static, relative, absolute, fixed)
    @mixin media .box_position_$(value)
        position: $value

/**
 * Place:
 *
 *  - box_place_*
 *
 **/

@each $value in (top, top-left, top-right, center, center-left, center-right, bottom, bottom-left, bottom-right)
    @mixin media .box_place_$(value)
        @if $value == top
            top: 0
            left: 50%
            transform: translateX(-50%)
        @else if $value == top-left
            top: 0
            left: 0
            transform: none
        @else if $value == top-right
            top: 0
            right: 0
            transform: none
        @else if $value == center
            top: 50%
            left: 50%
            transform: translate(-50%, -50%)
        @else if $value == center-left
            top: 50%
            left: 0
            transform: translateY(-50%)
        @else if $value == center-right
            top: 50%
            right: 0
            transform: translateY(-50%)
        @else if $value == bottom
            bottom: 0
            left: 50%
            transform: translateX(-50%)
        @else if $value == bottom-left
            bottom: 0
            left: 0
            transform: none
        @else if $value == bottom-right
            bottom: 0
            right: 0
            transform: none

/**
 * Margin:
 *
 *  - box_margin_*
 *
 **/

@each $value in (none, xxs, xs, s, m, l, xl, xxl, auto)
    @mixin media .box_margin_$(value)
        @if $value == none
            margin: 0
        @else if $value == auto
            margin: auto
        @else
            margin: var(--ek-box-indent-$(value))

    @mixin media .box_margin-tb_$(value)
        @if $value == none
            margin-top: 0
            margin-bottom: 0
        @else if $value == auto
            margin-top: auto
            margin-bottom: auto
        @else
            margin-top: var(--ek-box-indent-$(value))
            margin-bottom: var(--ek-box-indent-$(value))

    @mixin media .box_margin-lr_$(value)
        @if $value == none
            margin-left: 0
            margin-right: 0
        @else if $value == auto
            margin-left: auto
            margin-right: auto
        @else
            margin-left: var(--ek-box-indent-$(value))
            margin-right: var(--ek-box-indent-$(value))

    @mixin media .box_margin-left_$(value)
        @if $value == none
            margin-left: 0
        @else if $value == auto
            margin-left: auto
        @else
            margin-left: var(--ek-box-indent-$(value))

    @mixin media .box_margin-top_$(value)
        @if $value == none
            margin-top: 0
        @else if $value == auto
            margin-top: auto
        @else
            margin-top: var(--ek-box-indent-$(value))

    @mixin media .box_margin-right_$(value)
        @if $value == none
            margin-right: 0
        @else if $value == auto
            margin-right: auto
        @else
            margin-right: var(--ek-box-indent-$(value))

    @mixin media .box_margin-bottom_$(value)
        @if $value == none
            margin-bottom: 0
        @else if $value == auto
            margin-bottom: auto
        @else
            margin-bottom: var(--ek-box-indent-$(value))

/**
 * Padding:
 *
 *  - box_padding_*
 *
 **/

@each $value in (none, xxs, xs, s, m, l, xl, xxl)
    @mixin media .box_padding_$(value)
        @if $value == none
            padding: 0
        @else
            padding: var(--ek-box-indent-$(value))

    @mixin media .box_padding-tb_$(value)
        @if $value == none
            padding-top: 0
            padding-bottom: 0
        @else
            padding-top: var(--ek-box-indent-$(value))
            padding-bottom: var(--ek-box-indent-$(value))

    @mixin media .box_padding-lr_$(value)
        @if $value == none
            padding-left: 0
            padding-right: 0
        @else
            padding-left: var(--ek-box-indent-$(value))
            padding-right: var(--ek-box-indent-$(value))

    @mixin media .box_padding-left_$(value)
        @if $value == none
            padding-left: 0
        @else
            padding-left: var(--ek-box-indent-$(value))

    @mixin media .box_padding-top_$(value)
        @if $value == none
            padding-top: 0
        @else
            padding-top: var(--ek-box-indent-$(value))

    @mixin media .box_padding-right_$(value)
        @if $value == none
            padding-right: 0
        @else
            padding-right: var(--ek-box-indent-$(value))

    @mixin media .box_padding-bottom_$(value)
        @if $value == none
            padding-bottom: 0
        @else
            padding-bottom: var(--ek-box-indent-$(value))

/**
 * Round:
 *
 *  - box_round_*
 *
 **/

@each $value in (none, xxs, xs, s, m, l, xl, xxl, full)
    @mixin media .box_round_$(value)
        @if $value == none
            border-radius: 0
        @else if $value == full
            border-radius: 50%
        @else
            border-radius: var(--ek-box-round-$(value))

    @mixin media .box_round-top_$(value)
        @if $value == none
            border-top-left-radius: 0
            border-top-right-radius: 0
        @else if $value == full
            border-top-left-radius: 50%
            border-top-right-radius: 50%
        @else
            border-top-left-radius: var(--ek-box-round-$(value))
            border-top-right-radius: var(--ek-box-round-$(value))

    @mixin media .box_round-right_$(value)
        @if $value == none
            border-top-right-radius: 0
            border-bottom-right-radius: 0
        @else if $value == full
            border-top-right-radius: 50%
            border-bottom-right-radius: 50%
        @else
            border-top-right-radius: var(--ek-box-round-$(value))
            border-bottom-right-radius: var(--ek-box-round-$(value))

    @mixin media .box_round-bottom_$(value)
        @if $value == none
            border-bottom-left-radius: 0
            border-bottom-right-radius: 0
        @else if $value == full
            border-bottom-left-radius: 50%
            border-bottom-right-radius: 50%
        @else
            border-bottom-left-radius: var(--ek-box-round-$(value))
            border-bottom-right-radius: var(--ek-box-round-$(value))

    @mixin media .box_round-left_$(value)
        @if $value == none
            border-top-left-radius: 0
            border-bottom-left-radius: 0
        @else if $value == full
            border-top-left-radius: 50%
            border-bottom-left-radius: 50%
        @else
            border-top-left-radius: var(--ek-box-round-$(value))
            border-bottom-left-radius: var(--ek-box-round-$(value))

    @mixin media .box_round-top-left_$(value)
        @if $value == none
            border-top-left-radius: 0
        @else if $value == full
            border-top-left-radius: 50%
        @else
            border-top-left-radius: var(--ek-box-round-$(value))

    @mixin media .box_round-top-right_$(value)
        @if $value == none
            border-top-right-radius: 0
        @else if $value == full
            border-top-right-radius: 50%
        @else
            border-top-right-radius: var(--ek-box-round-$(value))

    @mixin media .box_round-bottom-right_$(value)
        @if $value == none
            border-bottom-right-radius: 0
        @else if $value == full
            border-bottom-right-radius: 50%
        @else
            border-bottom-right-radius: var(--ek-box-round-$(value))

    @mixin media .box_round-bottom-left_$(value)
        @if $value == none
            border-bottom-left-radius: 0
        @else if $value == full
            border-bottom-left-radius: 50%
        @else
            border-bottom-left-radius: var(--ek-box-round-$(value))


/**
 * Width:
 *
 *  - box_width_*
 *
 **/

@each $to in (1, 2, 3, 4, 5, 6, 7, 8, 9, 10)
    @each $from in (1, 2, 3, 4, 5, 6, 7, 8, 9, 10)
        @if $from < $to or ($from == 1 and $to == 1)
            @mixin media .box_width_$(from)-$(to)
                width: calc($from / $to * 100%)

@mixin media .box_width_auto
    width: auto

/**
 * Height:
 *
 *  - box_height_*
 *
 **/

@mixin media .box_height_1-1
    height: 100%

@mixin media .box_height_auto
    height: auto

/**
 * Overflow:
 *
 *  - box_overflow_*
 *  - box_overflow-x_*
 *  - box_overflow-y_*
 *
 **/

@each $value in (auto, hidden, scroll, visible)
    @mixin media .box_overflow_$(value)
        overflow: $value

    @mixin media .box_overflow-x_$(value)
        overflow-x: $value

    @mixin media .box_overflow-y_$(value)
        overflow-y: $value
