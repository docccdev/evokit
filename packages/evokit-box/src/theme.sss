@import './mixins.sss'

:root
    --ek-box-background-TEMPLATE_NAME: var(--background)
    --ek-box-border-TEMPLATE_NAME: var(--border)

@mixin media .box_background_TEMPLATE_NAME
    background: var(--ek-box-background-TEMPLATE_NAME)

/* BORDER */

@mixin media .box_border_TEMPLATE_NAME
    border: 1px solid var(--ek-box-border-TEMPLATE_NAME)

@mixin media .box_border-top_TEMPLATE_NAME
    border-top: 1px solid var(--ek-box-border-TEMPLATE_NAME)

@mixin media .box_border-right_TEMPLATE_NAME
    border-right: 1px solid var(--ek-box-border-TEMPLATE_NAME)

@mixin media .box_border-bottom_TEMPLATE_NAME
    border-bottom: 1px solid var(--ek-box-border-TEMPLATE_NAME)

@mixin media .box_border-left_TEMPLATE_NAME
    border-left: 1px solid var(--ek-box-border-TEMPLATE_NAME)

/* BORDER COLOR */

@mixin media .box_border-color_TEMPLATE_NAME
    border-color: var(--ek-box-border-TEMPLATE_NAME)

@mixin media .box_border-color-top_TEMPLATE_NAME
    border-top-color: var(--ek-box-border-TEMPLATE_NAME)

@mixin media .box_border-color-right_TEMPLATE_NAME
    border-right-color: var(--ek-box-border-TEMPLATE_NAME)

@mixin media .box_border-color-bottom_TEMPLATE_NAME
    border-bottom-color: var(--ek-box-border-TEMPLATE_NAME)

@mixin media .box_border-color-left_TEMPLATE_NAME
    border-left-color: var(--ek-box-border-TEMPLATE_NAME)

/* DEPRECATED */

@mixin media .box_border-tb_TEMPLATE_NAME
    border-top: 1px solid var(--ek-box-border-TEMPLATE_NAME)
    border-bottom: 1px solid var(--ek-box-border-TEMPLATE_NAME)

@mixin media .box_border-lr_TEMPLATE_NAME
    border-left: 1px solid var(--ek-box-border-TEMPLATE_NAME)
    border-right: 1px solid var(--ek-box-border-TEMPLATE_NAME)
