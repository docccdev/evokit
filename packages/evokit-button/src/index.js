import { createBlock } from 'evokit';

export const Button = createBlock('button', 'button', [
    'display',
    'events',
    'height',
    'size',
    'theme',
    'valign',
    'weight',
    'width',
    ['padding', [
        ['padding-top', 'padding-bottom'],
        ['padding-right', 'padding-left'],
        'padding-bottom',
        'padding-left',
    ]],
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
]);
