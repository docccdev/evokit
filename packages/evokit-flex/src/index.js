import { createBlock } from 'evokit';

export const Flex = createBlock('div', 'flex', [
    'direction',
    'display',
    'items',
    'wrap',
    ['content', [
        'content-align',
        'content-justify',
    ]],
]);

export const FlexItem = createBlock('div', 'flex__item', [
    'align',
    'display',
    'order',
]);

Flex.Item = FlexItem;
