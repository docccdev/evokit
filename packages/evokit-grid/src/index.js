import { createBlock } from 'evokit';

export const Grid = createBlock('div', 'grid', [
    'align',
    'column',
    'direction',
    'display',
    'divider',
    'divider-column',
    'height',
    'valign',
    'wrap',
    ['divider-indent', [
        'divider-indent-x',
        'divider-indent-y',
    ]],
    ['indent', [
        'indent-x',
        'indent-y',
    ]],
]);

export const GridItem = createBlock('div', 'grid__item', [
    'display',
    'order',
    'width',
]);

Grid.Item = GridItem;
