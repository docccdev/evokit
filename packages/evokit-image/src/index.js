import { createBlock } from 'evokit';

export const Image = createBlock('img', 'image', [
    'align',
    'alt',
    'border',
    'border-bottom',
    'border-left',
    'border-lr',
    'border-right',
    'border-tb',
    'border-top',
    'display',
    'fit',
    'mirror',
    'valign',
    'width',
    ['round', [
        ['round-top-left', 'round-bottom-right'],
        ['round-top-right', 'round-bottom-left'],
        'round-bottom-right',
        'round-bottom-left',
    ]],
    // DEPRECATED
    'round-bottom',
    'round-left',
    'round-right',
    'round-top',
]);
