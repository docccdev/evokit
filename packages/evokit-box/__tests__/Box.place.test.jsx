import React from 'react';
import { shallow } from 'enzyme';
import { Box } from '../src';

describe('<Box box-place="..." />', () => {
    it('props box-place', () => {
        const wrapper = shallow(<Box box-place='center' />);
        expect(wrapper).toMatchSnapshot();

        expect(wrapper.html()).toEqual('<div class="ek-box ek-box_place_center"></div>');
    });
});
