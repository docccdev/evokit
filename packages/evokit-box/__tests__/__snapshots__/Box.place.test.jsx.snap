// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box box-place="..." /> props box-place 1`] = `
<div
  className="ek-box ek-box_place_center"
/>
`;
